system/docker/client/Dockerfile

```diff
@@ -2,67 +2,75 @@ FROM teracy/ubuntu:16.04-dind-latest
 
 # generally useful packages
 RUN apt-get update && apt-get install -y --no-install-recommends \
-        ca-certificates \
-        apt-transport-https \
-        software-properties-common \
-        curl \
-        wget \
-        ssh \
-        vim \
+    ca-certificates \
+    apt-transport-https \
+    software-properties-common \
+    curl \
+    wget \
+    ssh \
+    vim \
     && rm -rf /var/lib/apt/lists/*
 
 # python
-ARG PYTHON3_VERSION
-ENV PYTHON3_VERSION=${PYTHON3_VERSION:-3.5}
```

Comment:
Could you please tell why the python version is being removed here? Soon it is planned to migrate to ubuntu20.04 and this variable may be needed due to the fact that the version of python will not coincide with the default version.
Of course, this is only relevant for the case when this Dockerfile will be edited to create an image for ubuntu20.04, and not a new one will be created from scratch.

Reply:
Right, the idea was to remove this since 3.5 is installed by default on xenial. You are also correct that if this was bumped to 20.04 that would cause this dockerfile to use 3.8 by default.

I think in the ideal case we use 3.8 if possible anyways when this is moved to 20.04, since I don't think we want to use 3.5 with 20.04 as it is no longer supported. So I think even with the update to 20.04 we would prefer to use the default python installation that comes with 20.04. If for some reason we aren't able to move to 3.8 we could add this back.

Is it desirable for someone running the tests to be able to switch python versions on the fly? In my experience so far it seems like everything is running on 3.5 currently, and with the move to 20.04 it seems likely that everything will run on 3.8. Is there a need for these tests to be run on arbitrary versions of python? If so I should probably put this back

Reply:
Given that we are using the default versions on both systems, I think you are right.  We can remove this variable.
I hope that we will write system tests for upgrading from Ubuntu 16.04 to Ubuntu 20.04 and there will be a change of python version on the fly, but the case is specific. Then we can return this variable if needed.